addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/handler/NetworkHandler.java

```diff
@@ -98,8 +94,12 @@ boolean confValueToBoolean(Object value) {
 
     @Override
     public void initialize() {
-        logger.debug("Initialize Network handler.");
-        this.networkService = new NetworkService();
+        // On each configuration change, we will be reinitialised. Stop the old network service first.
```

Comment:
The service actually needs to be stopped in dispose already. When initialize is called, you should always expect that nothing is running, so this check should not be done here.